src/api/v1/statuses.ts

```diff
@@ -111,12 +112,24 @@ function buildVisibilityConditions(viewerAccountId: Uuid | null | undefined) {
         ),
       ),
     ),
-    // Also include posts that mention the viewer
-    exists(
-      db
-        .select({ id: mentions.postId })
-        .from(mentions)
-        .where(eq(mentions.accountId, viewerAccountId)),
+    and(
+      eq(posts.visibility, "direct"),
```

Comment:
I think it should be applied to `private` posts as well!

```suggestion
      inArray(posts.visibility, ["private", "direct"]),
```